scene/3d/gpu_particles_3d.cpp

```diff
@@ -145,11 +145,23 @@ void GPUParticles3D::set_use_local_coordinates(bool p_enable) {
 }
 
 void GPUParticles3D::set_process_material(const Ref<Material> &p_material) {
+#ifdef TOOLS_ENABLED
+	if (process_material.is_valid()) {
+		if (cast_to<ParticleProcessMaterial>(*process_material)) {
```

Comment:
```suggestion
		if (Object::cast_to<ParticleProcessMaterial>(*process_material)) {
```
It's a static method